app/models/ability.rb

```diff
@@ -66,7 +66,13 @@ def signed_in(user)
 
     can [:new, :create], Registration do |registration|
       conference = registration.conference
-      conference.registration_open? && !conference.registration_limit_exceeded? || conference.program.speakers.confirmed.include?(user)
+      if conference.user_registered? user
+        false
```

Comment:
Would it make sense to add a `cannot` block for this case after this can block?

Reply:
@differentreality I didn't get what case are you taking about? But as far as i think you are taking about  **if conference.user_registered? user**.
I donot think why we should add a cannot block for this?

Reply:
As this can block is used to check wheather a user can register or not, so i think all the cases that depends on wheather a user can register or not should be in same block.